### PR TITLE
fix padding on pg str() function call expressions; cleanup STR() code

### PR DESCRIPTION
**Pull request by stephentgrammer**
fizzy#3422

### CODE REVIEW

File: src/translate/postgres.rs [hunk 1]
```diff
@@ -504,35 +504,29 @@ pub fn translate_fn_call<'parse, 'field_lookup>(
 
         // STOD(x) => COALESCE(TO_DATE(NULLIF(TRIM(x),''),'YYYYMMDD'),'0001-01-01')
         F::STOD => date("YYYYMMDD", argid(0)?, dst_tree),
-        // STR(num, len, dec) => PRINTF("%{len}.{dec}f", num)
+        // STR(num, len, dec) => LPAD(TOCHAR(num, "FM{len-dec-1}.0"), len, ' ')
```
Comment (jwhear) [inline src/translate/postgres.rs]: The comment here and the implementation don't match up. You're ignore `dec` altogether and emitting an `IIF`

Comment (stephentgrammer) [reply src/translate/postgres.rs]: clarified -- let me know what you think. I also added a  few more tests for edge cases.

**Comment by jwhear** [comment]
Blocked on #92

File: src/translate/postgres.rs [hunk 2]
```diff
@@ -504,35 +504,30 @@ pub fn translate_fn_call<'parse, 'field_lookup>(
...
-                        when_true: expression,
-                        when_false: dst_tree.push_expr(asterisks.into()),
-                    };
-                    ok(iif, FieldType::Character(len as u32))
-                }
-                StrArgs::WithoutArgs(val_arg) => {
-                    ok(Expression::Cast(val_arg, "text"), FieldType::Memo)
-                }
-            }
+            let (val_arg, fmt, len, _) = get_str_fn_args(args, src_tree, dst_tree, cx)?;
```
Comment (jwhear) [inline src/translate/postgres.rs]: Since the effect of `get_str_fn_args` now is to get the `len` and `dec` or their defaults and use the same format pattern regardless of how many args were passed, how about hoisting creating the `fmt` to this code. The SQLite version uses the same function but creates its own fmt, so this would move some work to the one spot where it's actually needed.

Comment (stephentgrammer) [reply src/translate/postgres.rs]: Good idea, done.
